tests/test_artifacts.py

```diff
@@ -0,0 +1,123 @@
+import os
+import responses
+
+from pyartifactory import ArtifactoryArtifact
+from pyartifactory.models import (
+    ArtifactPropertiesResponse,
+    ArtifactStatsResponse,
+    AuthModel,
+)
+
+URL = "http://localhost:8080/artifactory"
+AUTH = ("user", "password_or_apiKey")
+ARTIFACT_PATH = "my-repository/file.txt"
+ARTIFACT_NEW_PATH = "my-second-repository/file.txt"
+ARTIFACT_SHORT_PATH = "/file.txt"
+LOCAL_FILE_LOCATION = "tests/test_artifacts.py"
+ARTIFACT_PROPERTIES = ArtifactPropertiesResponse(
+    repo="my-repository", path=ARTIFACT_SHORT_PATH, createdBy="myself", uri="my_uri"
+)
+ARTIFACT_STATS = ArtifactStatsResponse(
+    uri="my_uri",
+    downloadCount=0,
+    lastDownloaded=0,
+    remoteDownloadCount=0,
+    remoteLastDownloaded=0,
+)
+
+
+class TestArtifact:
+    @staticmethod
+    @responses.activate
+    def test_deploy_artifact_success(mocker):
+        responses.add(responses.PUT, f"{URL}/{ARTIFACT_PATH}", status=200)
+
+        responses.add(
+            responses.GET,
+            f"{URL}/api/storage/{ARTIFACT_PATH}?properties[=x[,y]]",
+            json=ARTIFACT_PROPERTIES.dict(),
+            status=200,
+        )
+        artifactory = ArtifactoryArtifact(AuthModel(url=URL, auth=AUTH))
+        mocker.spy(artifactory, "properties")
+        artifact = artifactory.deploy(LOCAL_FILE_LOCATION, ARTIFACT_PATH)
+
+        artifactory.properties.assert_called_once_with(ARTIFACT_PATH)
+        assert artifact.dict() == ARTIFACT_PROPERTIES.dict()
+
+    @staticmethod
+    @responses.activate
+    def test_download_artifact_success():
+        artifact_name = ARTIFACT_PATH.split("/")[1]
+        responses.add(
+            responses.GET, f"{URL}/{ARTIFACT_PATH}", json=artifact_name, status=200
+        )
+
+        artifactory = ArtifactoryArtifact(AuthModel(url=URL, auth=AUTH))
+        artifact = artifactory.download(ARTIFACT_PATH)
+        os.remove("file.txt")
```

Comment:
You can use the "Temporary files & directories" fixtures from Pytest to avoid deleting files: https://docs.pytest.org/en/latest/tmpdir.html